### DISCUSSION TITLE
Git clients conflict

### DISCUSSION THREAD

**Discussion by ghost** [msys2/MSYS2-packages#4914] [General]
The repo is cloned by Git for Windows (Git patched to build with MinGW-w64?). When I tried to use the MSYS2 Git (Cygwin Git?) to do a `git pull`, I always see:

`error: Your local changes to the following files would be overwritten by merge:`

and

```
Please commit your changes or stash them before you merge.
Aborting
```

The problem is I didn't make any local changes. I can use Git for Windows to do a `git pull` without any problems.

Note: MSYS2 Git is a newer version compared to Git for Windows. I have not yet updated Git for Windows to the latest release. But I don't think this could a problem.

**Comment by jeremyd2019** [comment]
I think this behavior is usually due to differing line ending configurations between the git installs.  I'm not a git expert so I don't know the particular config name off the top of my head, but I think it is something like autoCRLF

**Reply by ghost** [reply]
I have also asked on their repository: https://github.com/git-for-windows/git/discussions/5163

Your answer seems to be more likely. The MSYS2 Git is Cygwin based so it will very likely use Unix line endings. The Git for Windows version is patched to build with MinGW-w64, and since it's a native Windows program it will very likely use Windows line endings.
